ntbs-service/Services/AuthorizationService.cs

```diff
@@ -55,18 +56,41 @@ public async Task<bool> CanEditBannerModelAsync(ClaimsPrincipal user, Notificati
             var tbServiceCode = notificationBannerModel.TbServiceCode;
             var tbServicePhecCode = notificationBannerModel.TbServicePHECCode;
             var locationPhecCode = notificationBannerModel.LocationPHECCode;
-            return await AuthorizeUserAccess(user, tbServiceCode, locationPhecCode, tbServicePhecCode);
+            return await AuthorizeUserAccessForTbServiceAndPhecCodes(user, tbServiceCode, locationPhecCode, tbServicePhecCode);
+        }
+
+        public async Task<PermissionLevel> GetPermissionLevelForNotificationAsync(ClaimsPrincipal user,

```

Comment:
Would appreciate some feedback on the best way to organise these functions - had a brief chat with Jan about this before but wasn't quite able to implement what was explained.

Current logic is that GetPermissionLevel performs a check for Edit level permission followed by readonly or defaults to none. 

It doesn't feel great to then still have CanEditNotificationAsync and a function that both combines getting the filter and filtering a bunch of codes

Reply:
It's definitely getting there!
Here I think it's useful to think about which bit of the code is being the most problematic, and it seems that the `CanEditNotificationAsync` / `AuthorizeUserAccessForTbServiceAndPhecCodes` pair of functions is being a bit of a rigid thorn. As the first red flag, their names are referring to business logic (edit / access), rather than to the checks that they are performing. Perhaps we want to break it up a little further and, imagine the "target" `GetPermissionLevelForNotificationAsync` function, and go from there. An example of that could be
```
PermissionLevel GetPermissionLevelForNotificationAsync(...) {
if (UserHasDirectRelationToNotification)
  return Edit
else if (UserHasDirectRelationToLinkedNotification)
 return ReadOnly
else 
  return None
}

bool UserHasDirectRelationToNotification = userBelongsToTbServiceOfNotification || userBelongsToTreatmentPhecOfNotification || userBelonsToResidencePhecOfNotification

bool UserHasDirectRelationToLinkedNotification = ...
```